tests/test_sparse.py

```diff
@@ -0,0 +1,53 @@
+import random
+
+import jax.numpy as jnp
+import pytest
+
+import dynamiqs as dq
+
+
+class TestSparseDIA:
+    @pytest.fixture(autouse=True)
+    def _setup(self):
+        N = 4
```

Comment:
I would be tempted to set `N` to slightly larger than `4` to make sure we are not lucky. Something like 6 or 10 should be plenty.